### PR TITLE
added account info registration step implementation

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/Core/Registration/RegistrationModel.cs [hunk 1]
```diff
@@ -6,54 +6,81 @@
...
+        public string LastName { get; private set; }
+        public string CountryCodeIso2 { get; private set; }
+        public string PhoneNumber { get; private set; }
+
         public RegistrationModel(string email)
         {
             Email = email;
-            RegistrationId = GenerateRegistrationId();
+            ActiveStep = RegistrationStep.InitialInfo;
+            RegistrationId = Convert.ToBase64String(Guid.NewGuid().ToByteArray())
```
Comment (simpllfy) [inline src/Core/Registration/RegistrationModel.cs]: Private method was improving readability of this code. Why did you remove it?

File: src/Core/Registration/RegistrationModel.cs [hunk 2]
```diff
@@ -6,54 +6,81 @@
...
+                .Replace("/", "_")
+                .Replace("+", "-")
+                .Substring(0, 22);
         }
 
-        [SerializationConstructor]
-        public RegistrationModel(string registrationId, string email, string clientId, string hash, string salt, RegistrationStep registrationStep)
+        public void CompleteStep(object context)
```
Comment (simpllfy) [inline src/Core/Registration/RegistrationModel.cs]: Generic method will lead to:
- types casting (you need to cast 'context' somewhere inside)
- increasing cognitive complexity (you need to go into method, and lookup the code within switch construction)
- unit tests writing also will be more complex - because you need to test single method, so it will be quite hard to manage all of them

I suggest to split this one into N methods, one for each step, with meaningful names.

File: src/Core/Registration/RegistrationModel.cs [hunk 3]
```diff
@@ -6,54 +6,81 @@
...
         {
-            if (initialInfoDto.Email != Email)
-                throw new ArgumentException("Email doesn't match to verified one.");
-            if (!IsPasswordComplex(initialInfoDto.Password))
-                throw new PasswordIsNotComplexException();
-
-            ClientId = initialInfoDto.ClientId;
-            this.SetPassword(initialInfoDto.Password);
-            RegistrationStep = RegistrationStep.AccountInformation;
+            return phoneNumber.PreparePhoneNum().ToE164Number() != null;
```
Comment (gmaf) [inline src/Core/Registration/RegistrationModel.cs]: Better to add ? null check
```suggestion
            return phoneNumber?.PreparePhoneNum()?.ToE164Number() != null;
```

File: src/Core/Registration/RegistrationModel.cs [hunk 4]
```diff
@@ -6,54 +6,81 @@
...
-            RegistrationStep = registrationStep;
-        }
+            switch (ActiveStep)
+            {
+                case RegistrationStep.InitialInfo:
+                {
+                    if (context is InitialInfoDto ctx)
+                    {
+                        if (ctx.Email != Email)
+                            throw new ArgumentException("Email doesn't match to verified one.");
```
[the remark this answers is not in the document]
Comment (tarurar) [reply src/Core/Registration/RegistrationModel.cs]: Fixed

File: src/Core/Registration/RegistrationModel.cs (hunk 3 above)
Comment (tarurar) [reply src/Core/Registration/RegistrationModel.cs]: Fixed

File: src/Core/Registration/RegistrationModel.cs [hunk 5]
```diff
@@ -6,54 +6,73 @@
...
+
+            FirstName = context.FirstName;
+            LastName = context.LastName;
+            CountryOfResidenceIso2 = context.CountryCodeIso2;
+            PhoneNumber = context.PhoneNumber;
 
-            ClientId = initialInfoDto.ClientId;
-            this.SetPassword(initialInfoDto.Password);
-            RegistrationStep = RegistrationStep.AccountInformation;
+            CurrentStep += 1;
```
Comment (simpllfy) [inline src/Core/Registration/RegistrationModel.cs]: I suggest to set here concrete value without relying on enum sequence

File: src/Core/Registration/RegistrationModel.cs [hunk 6]
```diff
@@ -6,54 +6,73 @@
...
+
+            if (context.Email != Email)
+                throw new RegistrationEmailMatchingException(context.Email);
+            if (!IsPasswordComplex(context.Password))
+                throw new PasswordIsNotComplexException();
+
+            ClientId = context.ClientId;
+            this.SetPassword(context.Password);
+
+            CurrentStep += 1;
```
Comment (simpllfy) [inline src/Core/Registration/RegistrationModel.cs]: I suggest to set here concrete value without relying on enum sequence
